frontend/src/types/types.ts

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { ActionMeta, OptionsType } from "react-select";
+
+export type Data = {
+  clicks: number;
+  impressions: number;
+  date: string;
+  datasource: string;
+  campaign: string;
+};
+
+export type DataSource = {
+  value: string;
+};
+
+export type Campaign = {
+  name: string;
+  label: string;
+  index: number;
+};
+
+export type CampaignOptions = {
+  value: string;
+  label: string;
+};
+
+export type OnInputChange = (
+  event: React.ChangeEvent<HTMLInputElement>
+) => void;
+
+export type OnSelectChange = (
+  value: OptionsType<{
```

Comment:
These look like `SelectOption` any reason not to use it?

Reply:
I missed that somehow ; ) fixing that and other places.